### DISCUSSION TITLE
`:format` does not work in unsaved buffers

### DISCUSSION THREAD

**Discussion by gruhn** [helix-editor/helix#12024] [Troubleshooting]
### Summary

Nothing terrible, but I noticed that the `:format` command doesn't seem to do anything in unsaved buffers. Sometimes I just want to pretty print some JSON, without saving it anywhere. 

### Reproduction Steps

1. Open helix and paste some unformatted JSON into the unsaved buffer (e.g. `{ "json_key": [1,2,3] }`)
2. Type `:set-language json`
3. Type `:format` (no effect)
4. Instead save the buffer with `:w temp.json` 
5. Type `:format` (now it works)

https://github.com/user-attachments/assets/7a8cd1b0-f3c8-4a01-b79a-46cd220926d8

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

macOS

### Terminal Emulator

iTerm2

### Installation Method

nixpkgs

### Helix Version

helix 24.7 (079f5442)

**Comment by the-mikedavis** [comment]
This is a limitation of LSP: LSP features only work on buffers that are backed by files. It should work to define an external formatter (`formatter` key in https://docs.helix-editor.com/master/languages.html#language-configuration) like prettier, since that works without an underlying file and takes precedence over LSP formatting.
